File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.ts
function sumAndMultiply(n: number): number {
    let x = 0;
    let sum = 0;

    // 문자열로 변환하여 왼쪽부터 순회
    const str: string = n.toString();

    for (const ch of str) {
        if (ch !== "0") {
            const digit: number = Number(ch);
            sum += digit;
            x = x * 10 + digit;
        }
    }

    return x * sum;
}

File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.java
class Solution {
    public long sumAndMultiply(int n) {
        long x = 0;
        long sum = 0;
        
        // Convert to string to easily traverse left-to-right
        String str = String.valueOf(n);
        
        for (char ch : str.toCharArray()) {
            if (ch != '0') {
                int digit = ch - '0';
                sum += digit;
                x = x * 10 + digit;
            }
        }
        
        return x * sum;
    }
}
